src/mmw/apps/modeling/views.py

```diff
@@ -254,22 +256,29 @@ def start_mapshed(request, format=None):
 
 
 def _initiate_mapshed_job_chain(mapshed_input, job_id):
+    workers = choose_workers()
+    get_worker = lambda: random.choice(workers)
     errback = save_job_error.s(job_id).set(exchange=MAGIC_EXCHANGE,
-                                           routing_key=choose_worker())
+                                           routing_key=get_worker())
 
     geom = GEOSGeometry(json.dumps(mapshed_input['area_of_interest']),
                         srid=4326)
 
-    chain = (group(geop_tasks(geom, errback, MAGIC_EXCHANGE, choose_worker)) |
-             combine.s().set(exchange=MAGIC_EXCHANGE,
-                             routing_key=choose_worker()) |
-             collect_data.s(geom.geojson).set(link_error=errback,
-                                              exchange=MAGIC_EXCHANGE,
-                                              routing_key=choose_worker()) |
-             save_job_result.s(job_id, mapshed_input)
-             .set(exchange=MAGIC_EXCHANGE, routing_key=choose_worker()))
+    job_chain = []
```

Comment:
Is this switch from the pipe-syntax to creating an array and chaining at the end the main thing that provides the speed up?